### ISSUE TITLE
Integrate Simorgh Gist Component to Optimo Article

### ISSUE THREAD

**Issue by JonBeeb** [bbc/simorgh#9258]
**Is your feature request related to a problem? Please describe.**
Give a users the Gist of an article before they commit to it.
A condensed version of a longer form article that users can consume at a glance

### User story
As a maximiser, Sareh
I want to get the gist of an article
so that I understand what it's about without having to read the full article

Notes taken from : NEWSWORLDSERVICE-1249
Zeplin designs: https://zpl.io/agmxjR0
Screen reader UX: https://zpl.io/29GOXxd
![375_Mundo gist](https://user-images.githubusercontent.com/16046375/126348074-24e69259-cfc8-4f62-964a-4d6407fc564a.png)
![375_Mundo gist without photo](https://user-images.githubusercontent.com/16046375/126348091-9e90afaa-ca31-4068-b6af-9a9c22f8cdc7.png)

PS version of the Gist in storybook: https://static.files.bbci.co.uk/core/storybook/index.html?path=/story/web-core-containers-article--optimo-with-gist&knob-Lever:%20brand%20palette=news&knob-Lever:%20core%20palette=light&knob-Lever:%20font%20palette=sansSimple

**Describe the solution you'd like**
Integrate the gist component created in #9279 into the optimo article.

### Miro
All the following requirement are also miro
https://miro.com/app/board/o9J_l8ojnGE=/
![image](https://user-images.githubusercontent.com/16046375/126348873-38d5ed2e-49c6-4657-84f4-32428be354c0.png)

### Contents of the Gist
The component consists of 4 parts:
**Red line**
This should be shown at the top of the Gist. This will only be shown on WS/news PS optimo articles, so does not need to be configurable

**Background**
The background of the Gist is white. This option should be independent of the main background colour. (There is a separate ticket for changing the background of the article to the grey colour.)

**Title**
The block title: English 'At a glance'
This localised text for each service will be picked up from the site config using key of gist

If a localised version is not available, show the default English title

**Bullet point list**
This will use the current bullet list component
The content of the bullet list will depend on editorial for number of bullet points and length of each entry.

NOTE: UX shows bullet points as 'square'. They may start off as 'round' depending on if we use the bullet point component

**Translations**
Need to translate the phrase 'At a glance' in to all languages
If no translation is available for that service, we will use a default value supplied by editorial.

### When to show a Gist

The gist should be shown on
    On any WS optimo article page (canonical and amp)
    On any PS optimo news amp page

The Gist is an editorially added piece of content and editorial conditions need to be met to display the Gist section on the page

**Describe alternatives you've considered**

**Testing notes**
### Test urls

Optimo | Page
-- | --
WS Gist test Multiple Overviews - Pidgin | https://www.test.bbc.com/pidgin/articles/cvrvly5wkyqo
WS Gist test with other Overview blocks - Pidgin | https://www.test.bbc.com/pidgin/articles/cr8ng5v7lr9o
Untitled (no H1) | https://www.test.bbc.com/pidgin/articles/c891ke22dllo
WS Gist test without image - Pidgin | https://www.test.bbc.com/pidgin/articles/ce2pze0xlrpo
WS Gist test with image - Pidgin | https://www.test.bbc.com/pidgin/articles/cdj9pxy719jo

### Acceptance Criteria

**Showing a gist**
Given a journalist has embedded a gist block within an Optimo article
when a user visits the article
then the gist block is displayed to the user
and the gist block is at the top of the page
and the gist block contains a bullet point list
and the gist block displays Bold test with the label "At a glance" in the correct language and font of the service
and the gist block is aligned RTL for RTL services

**Contents of Gist**
Gist Title
Given I am looking at the Gist block on an Optimo article on the WS
When I look at the title of the block
Then the text should be the value configured for that service

Gist bar
Given I am looking at the Gist block on an Optimo article on the WS
When I look at the colour of the line at the top of the block
Then it should always be news red

Gist background
Given I am looking at the Gist block on an Optimo article on the WS
When I look at the colour of the background of the block
Then It should be white
And may be different from the article background colour

Gist bullet points
Given I am looking at an Optimo article on a World Service page
When I look at the content on the bullet points
Then it should match what is entered in Optimo
And no checks will be made by the platform

**When to show a Gist**
Showing a Gist on WS optimo article is optional.
The following conditions all must be true to show the Gist

Showing Gist on WS optimo pages
Given I am on a World Service Canonical or AMP optimo page
When I render the page
Then check if the article has a Gist

Showing Gist on PS optimo AMP pages
 Given I am on a PS News AMP page
When I render the page
Then check if the article has a Gist

Matching editorial conditions to show a Gist
Given I am on a Word Service or News Optimo article page
When the ares feed contains a group block with type = 'group' - "type": "group",
And the 'group block' is set to "overview" - "intentType": "overview"
And the first block is a text block - "type": "text"
And the block contains an unordered list - type": "unorderedList",
And the list has at least 1 entry - "type": "listItem"
Then show the Gist block on the page showing the unordered list
Else Don't show the block

(From https://github.com/bbc/web/issues/4883)
 
**Positioning the Gist**
Optimo allows blocks to be added anywhere on a page, but the Gist needs to be set in a specific position.
If the rules for displaying a gist are met (anywhere within the article) then position the Gist as follows

No Headline block on mobile or desktop
Given you are an an WS Optimo article page
When there is no Headline block
Then show the Gist

Headline block on mobile or desktop
Given you are an an WS Optimo article page
When there is a Headline block
AND the next block is not a media or Image Block
Then show the Gist after the Headline

Headline followed by an Image or Media block on mobile or desktop
Given you are on an WS Optimo article page
When there is a Headline block
AND the next block is a media or Image Block
Then show the Gist after the media or Image block
 
### a11y A/C

**Screen reader**
Given I use a screen reader
When I navigate to the Gist block on an article page
Then a region landmark is announced (e.g. Region)
And this must be announced with a label
And the label must be the same as the strong HTML element (i.e. the block title "At a glance")
And this landmark sits within the main landmark

Screen reader UX
Given I am using a screen reader
When I navigate to the Gist block on an article page
Then this should not have heading semantics
And the block title (i.e. "At a glance") should be a strong HTML element*

Given I use a screen reader
When there is more than one item
Then list semantics are announced
And each item is a list item within the unordered list

Given I am using the VoiceOver screen reader on iPhone and Mac
When I navigate to the unordered list
Then the content should be announced along with list semantics
And nothing else should be announced e.g. It should not give a hint of the visual look of the list bullet

**Colour contrast**

Given I am viewing the Gist block
When content is displayed
Then the content and the line/border at the top of the block must meet the colour contrast ratio of 4.5:1 or higher

Given I am viewing an article with a Gist block in Firefox
When I change the colour preferences
And I customise the colours
Then the Gist block must be visible
And a line/border must be seen at the top of the block
And a line/border must be seen around the component
And a 'bullet' must be seen for all list items

Given I am viewing an article with a Gist block on Windows with IE11 and Edge
When I select a High Contrast Mode
And I customise the colours
Then the Gist block must be visible
And be in the customised colours selected
And a line/border must be seen at the top of the block
And a line/border must be seen around the component
And a 'bullet' must be seen for all list items

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

**Comment by ryanmccombe** [comment]
This is on test now - needs QA

**Comment by paruchurisilpa** [comment]
Done cross browser and Gist on Optimo articles LTGM..

**Comment by rebeccamcginn** [comment]
Merged
